### ISSUE TITLE
Image encoding is a bottleneck - Implement multithreaded image writing?

### ISSUE THREAD

**Issue by n00mkrad** [hzwer/arXiv2020-RIFE#95]
Using uncompressed BMP instead of PNG increased my speed from ~14 FPS at 1080p to ~18 FPS, proving that the image encoding slows things down and bottlenecks the GPU.

Is it possible to run image writing in several threads to speed this up?

**Comment by talosh** [comment]
True, I'm doing it with multiprocessing for sequence processing with 16bit fp open-exr, cv2 supports only zip compression and  is quite heavy. Big improvement with giving it to threads like this: 
path = os.path.join(os.path.abspath(args.output), '{:0>7d}.exr'.format(cnt))
p = mp.Process(target=cv2.imwrite, args=(path, item[:, :, ::-1], [cv2.IMWRITE_EXR_TYPE, cv2.IMWRITE_EXR_TYPE_HALF], ))
p.start()

full code: https://github.com/talosh/flameTimewarpML/blob/main/bundle/inference_sequence.py
